From 97cb3288a496b6ccff63c8ac69580b67f3dfce60 Mon Sep 17 00:00:00 2001
From: Chen Jiahao <chenjiahao16@huawei.com>
Date: Wed, 21 Apr 2021 10:32:59 +0800
Subject: [PATCH] arm64: fix USER_DS definition problem in non-compat mode
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 97cb3288a496b6ccff63c8ac69580b67f3dfce60


hulk inclusion
category: bugfix
bugzilla: 51788
CVE: NA

--------

For some platforms which CONFIG_COMPAT is not set, the function 'is_compat_task'
is not declared. Therefore, the macro USER_DS cannot be defined using the function
'is_compat_task' in this case.

Fixes: 2ef73d5148e3 ("arm64: fix current_thread_info()->addr_limit setup")
Signed-off-by: Chen Jiahao <chenjiahao16@huawei.com>
Reviewed-by: Chang Liao <liaochang1@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/include/asm/processor.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/include/asm/processor.h b/arch/arm64/include/asm/processor.h
index f28a8048bcf0..513d625f2267 100644
--- a/arch/arm64/include/asm/processor.h
+++ b/arch/arm64/include/asm/processor.h
@@ -9,8 +9,12 @@
 #define __ASM_PROCESSOR_H
 
 #define KERNEL_DS		UL(-1)
+#ifdef CONFIG_COMPAT
 #define USER_DS			(is_compat_task() ? \
 				(UL(0x100000000) - 1) : (TASK_SIZE - 1))
+#else
+#define USER_DS			(TASK_SIZE - 1)
+#endif
 
 /*
  * On arm64 systems, unaligned accesses by the CPU are cheap, and so there is
-- 
2.26.2

